Add Mock for Mock Smm Sw Dispatch2 Protocol

Rasheed-Yusuf opened the pull request:
## Description

Add Mock for Mock Smm Sw Dispatch2 Protocol

For details on how to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [x] Backport to release branch?

## How This Was Tested

Incorporated into GoogleTest

## Integration Instructions

N/A

Rasheed-Yusuf committed:
Add mock protocol under MdePkg.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockFirmwareVolumeBlock.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Rasheed-Yusuf committed:
Add mock protocol under MdePkg.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockSmmSwDispatch2.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Rasheed-Yusuf committed:
Add mock protocol under MdePkg.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockSmmSwDispatch2.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:
```diff
@@ -0,0 +1,103 @@
+  This file declares a mock of Firmware Volume Block Protocol
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
This should be replaced with the SPDX license identifier in all files.

```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h:
```diff
@@ -0,0 +1,56 @@
+  This file declares a mock of SMM Software Dispatch Protocol
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

makubacki commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h:
Did you intend to add the firmware volume block mock to both this PR and https://github.com/microsoft/mu_basecore/pull/1216?

Rasheed-Yusuf replied:
No. That was unintended.

Rasheed-Yusuf committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h

Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h

Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Delete MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h

This file was added inadvertently.

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Modifications for CI build errors.

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockSmmSwDispatch2.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Rasheed-Yusuf committed:
CI build formatting fixes

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h:
```diff
@@ -0,0 +1,55 @@
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockRegister, 4, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockSmmSwDispatch2Protocol, MockUnRegister, 2, EFIAPI);
+
+EFI_SMM_SW_DISPATCH2_PROTOCOL  SMM_SW_DISPATCH2_PROTOCOL_MOCK = {
```
Based on the latest discussions in Edk2, the definition of MockEfiSmmSwDispatch2Protocol  should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9

VivianNK commented on MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h:
```diff
@@ -0,0 +1,55 @@
+}
+
+// Declarations to handle usage of the EFI_SMM_SW_DISPATCH2_PROTOCOL
+struct MockSmmSwDispatch2Protocol {
```
```suggestion
struct MockEfiSmmSwDispatch2Protocol {

```

Rasheed-Yusuf committed:
Update MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockSmmSwDispatch2.h

Co-authored-by: Vivian Nowka-Keane <vnowkakeane@microsoft.com>
Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockSmmSwDispatch2.h

Modifications to use #define MOCK_EFI_SMM_SW_DISPATCH2_PROTOCOL_INSTANCE for MockEfiSmmSwDispatch2Protocol definition

Mock Protocol:
MdePkg\Test\Mock\Include\GoogleTest\Protocol\MockSmmSwDispatch2.h

Signed-off-by: Rasheed Yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>

Rasheed-Yusuf committed:
Update MockSmmSwDispatch2.h to fix CI build errors

Signed-off-by: Rasheed-Yusuf <149545750+Rasheed-Yusuf@users.noreply.github.com>
